src/Simplex/Chat/Store.hs

```diff
@@ -281,12 +310,233 @@ getConnectionChatDirection st userId agentConnId =
           (userId, contactId)
     toContact contactId c [(localContactRef, contactRef, displayName)] =
       let profile = Profile {contactRef, displayName}
-       in Right Contact {contactId, localContactRef, profile, activeConn = c}
+       in Right $ CContact Contact {contactId, localContactRef, profile, activeConn = c}
     toContact _ _ _ = Left $ SEInternal "referenced contact not found"
 
+-- | creates completely new group with a single member - the current user
+createNewGroup :: (MonadUnliftIO m, MonadError StoreError m) => SQLiteStore -> TVar ChaChaDRG -> User -> GroupProfile -> m Group
+createNewGroup st gVar User {userId, userContactId, profile} p@GroupProfile {groupRef, displayName} =
+  liftIOEither . checkConstraint SEDuplicateGroupRef . withTransaction st $ \db -> do
+    -- group inserted before profile to ensure its local_group_ref is unique
+    DB.execute db "INSERT INTO groups (local_group_ref, lgr_base, user_id) VALUES (?, ?, ?)" (groupRef, groupRef, userId)
+    groupId <- insertedRowId db
+    DB.execute db "INSERT INTO group_profiles (group_ref, display_name) VALUES (?, ?)" (groupRef, displayName)
+    profileId <- insertedRowId db
+    DB.execute db "UPDATE groups SET group_profile_id = ? WHERE group_id = ?" (profileId, groupId)
+    memberId <- randomId gVar 12
+    createMember_ db groupId userContactId GROwner GSMemReady (Just userContactId) memberId
+    groupMemberId <- insertedRowId db
+    let membership =
+          GroupMember
+            { groupMemberId,
+              memberId,
+              memberRole = GROwner,
+              memberStatus = GSMemReady,
+              invitedBy = IBUser,
+              memberProfile = profile,
+              memberContactId = Just userContactId
+            }
+    pure $ Right Group {groupId, localGroupRef = groupRef, groupProfile = p, members = [], membership}
+
+-- | creates a new group record for the group the current user was invited to
+createGroup :: (MonadUnliftIO m, MonadError StoreError m) => SQLiteStore -> TVar ChaChaDRG -> User -> Contact -> GroupProfile -> m Group
+createGroup st gVar User {userId, userContactId, profile} contact p@GroupProfile {groupRef, displayName} =
+  liftIOEither . withTransaction st $ \db -> do
+    DB.execute db "INSERT INTO group_profiles (group_ref, display_name) VALUES (?, ?)" (groupRef, displayName)
+    profileId <- insertedRowId db
+    lgrSuffix <- getLgrSuffix db
+    group <- create db profileId lgrSuffix 20
+    pure group
+  where
+    -- createMember_ db groupId userContactId GROwner GSMemReady (Just userContactId) memberId
```

Comment:
this is commented because user is not initially a member of the group he is invited to?

Reply:
just not complete yet